public/components/DataTable/AlertingDataTable.tsx

```diff
@@ -0,0 +1,408 @@
+ */
+
+import './AlertingDataTable.scss';
+import React, { useState, useMemo } from 'react';
+import {
+  EuiSmallButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiCodeBlock,
+  EuiTabbedContent,
+  EuiText,
+  EuiInMemoryTable,
+  EuiPagination,
+  EuiCallOut,
+  EuiSmallButtonEmpty,
+  EuiTextColor,
+} from '@elastic/eui';
+import { FormattedMessage } from '@osd/i18n/react';
+import { i18n } from '@osd/i18n';
+import dompurify from 'dompurify';
+
+interface AlertingDataTableProps {
+  pplResponse: any;
+  isLoading?: boolean;
+  services: any;
+}
+
+const PAGINATED_PAGE_SIZE = 5;
+
+/**
+ * Flatten nested objects recursively (like Discover does)
+ */
+const flattenObject = (obj: any, prefix = '', out: Record<string, any> = {}): Record<string, any> => {
+  if (obj == null) return out;
+  
+  Object.keys(obj).forEach((k) => {
+    const key = prefix ? `${prefix}.${k}` : k;
+    const val = obj[k];
+    
+    if (val && typeof val === 'object' && !Array.isArray(val)) {
+      // Recursively flatten nested objects
+      flattenObject(val, key, out);
+    } else if (Array.isArray(val)) {
+      // Handle arrays - show as comma-separated values
+      out[key] = val.map(v => 
+        typeof v === 'object' ? JSON.stringify(v) : String(v)
+      ).join(', ');
+    } else {
+      // Handle primitives
+      out[key] = val;
+    }
+  });
+  
+  return out;
+};
+
+/**
+ * Convert PPL response to flattened documents
+ */
+const pplResponseToDocs = (pplResponse: any) => {
+  if (!pplResponse || !Array.isArray(pplResponse.schema) || !Array.isArray(pplResponse.datarows)) {
+    return [];
+  }
+
+  const { schema, datarows } = pplResponse;
+  
+  return datarows.map((row, index) => {
+    const doc: Record<string, any> = {};
+    schema.forEach((col: any, idx: number) => {
+      doc[col.name] = row[idx];
+    });
+    
+    // Flatten the document like Discover does
+    const flattened = flattenObject(doc);
+    return { _id: `${index}`, ...flattened };
+  });
+};
+
+/**
+ * Stream Preview Row - Shows compact log entry like Discover
+ */
+const StreamPreviewRow: React.FC<{
+  doc: Record<string, any>;
+  columns: string[];
+  index: number;
+}> = ({ doc, columns, index }) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  // Create a compact preview string like Discover's stream view with highlighted field names
+  const previewText = Object.entries(doc)
+    .filter(([k]) => k !== '_id')
+    .map(([k, v]) => {
+      if (v === null || v === undefined) return '';
+      const value = typeof v === 'number' ? v.toLocaleString() : String(v);
+      return (
+        <span key={k}>
+          <span className="field-name-highlight">{k}</span>:{value}
+        </span>
+      );
+    })
+    .filter(Boolean);
+
+  const expandedContent = () => {
+    // Sort fields to match Discover's order (put _id, _index, _score, _type at the end)
+    const sortedEntries = Object.entries(doc)
+      .filter(([k]) => k !== '_id')
```

Comment:
Do we have an idea of how many entries could be in the doc? Calling `filter` followed by `sort` is essentially iterating through 2 arrays that have overlapping values. If we're only filtering out `_id` entries (which I imagine we're expecting that there will only be 1 `_id` entry, correct?), this would end up iterating through an array with _n_  entries, and then iterate through basically the same array execept with _n_ - 1 entries. This might not be a big deal if we're not expecting a lot of entries, but this could be optimized.